### PR TITLE
fix(JsonFormatter): type the constructor

### PR DESCRIPTION
**Pull request by erdnaxeli**
Now that the package include `py.typed`, mypy actually checks the type. I am using `disallow-untyped-calls`, so it fails because the constructor of `JsonFormatter` is untyped. This PR fixes it.

I run `tox` which runs `black`, which fixed some formatting issues.

### CODE REVIEW

File: src/pythonjsonlogger/jsonlogger.py [hunk 1]
```diff
@@ -89,22 +106,34 @@ class JsonFormatter(logging.Formatter):
...
+        json_serialiser: OptionalCallableOrStr = json.dumps,
+        json_indent: Optional[Union[int, str]] = None,
+        json_ensure_ascii: bool = True,
+        prefix: str = "",
+        rename_fields: Optional[dict] = None,
+        static_fields: Optional[dict] = None,
+        reserved_attrs: Tuple[str, ...] = RESERVED_ATTRS,
+        timestamp: Union[bool, str] = False,
+        **kwargs: Any
+    ):
```
Comment (erdnaxeli) [inline src/pythonjsonlogger/jsonlogger.py:123]: The important part is here.

File: src/pythonjsonlogger/jsonlogger.py [hunk 2]
```diff
@@ -113,26 +142,24 @@ def __init__(self, *args, **kwargs):
...
+        self.json_encoder = self._str_to_fn(json_encoder)
+        self.json_serializer = self._str_to_fn(json_serialiser)
+        self.json_indent = json_indent
+        self.json_ensure_ascii = json_ensure_ascii
+        self.prefix = prefix
+        self.rename_fields = rename_fields or {}
+        self.static_fields = static_fields or {}
         self.reserved_attrs = dict(zip(reserved_attrs, reserved_attrs))
-        self.timestamp = kwargs.pop("timestamp", False)
+        self.timestamp = timestamp
```
Comment (erdnaxeli) [inline src/pythonjsonlogger/jsonlogger.py:154]: and here

File: src/pythonjsonlogger/jsonlogger.py [hunk 3]
```diff
@@ -89,22 +106,34 @@ class JsonFormatter(logging.Formatter):
...
         """
         :param json_default: a function for encoding non-standard objects
             as outlined in https://docs.python.org/3/library/json.html
         :param json_encoder: optional custom encoder
         :param json_serializer: a :meth:`json.dumps`-compatible callable
             that will be used to serialize the log record.
-        :param json_indent: an optional :meth:`json.dumps`-compatible numeric value
-            that will be used to customize the indent of the output json.
+        :param json_indent: indent parameter for json.dumps
+        :param json_ensure_ascii: ensure_ascii parameter for json.dumps
```
Comment (erdnaxeli) [inline src/pythonjsonlogger/jsonlogger.py:131]: I removed the duplicated doc for `json_indent` and I moved `json_ensure_ascii` next to it.

**Review by madzak — COMMENTED** [review]
As an aside, typing really makes this code look ugly as hell :(

**Review by madzak — APPROVED** [review]

**Comment by jenstroeger** [comment]
> As an aside, typing really makes this code look ugly as hell :(

Heheh, but it’s worthwhile to catch bugs. But anyway, do you know when this will roll out into a new release?

**Comment by erdnaxeli** [comment]
Hi, could you consider making a new release? Currently all our apps have a version constraint `<2.7` on this lib in order to mypy to pass.

**Comment by OmerFI** [comment]
Hi @madzak could you publish a release with this change included?

**Pull request merged**